core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/AST/NodeKindPropertyShape.java

```diff
@@ -36,7 +36,7 @@
 		super(id, connection, nodeShape, deactivated, parent, path);
 
 		this.nodeKind = NodeKind.from(nodeKind);
-
+		this.Expected = nodeKind;
```

Comment:
Seeing that every property shape needs to set this variable I'm starting to think that maybe we should have a method in the interface called getExpected() instead. This way there would be no way to forget to set the variable.